### PR TITLE
Add instructions on how to deploy docs pages to github pages

### PR DESCRIPTION
**Pull request by nvaytet**
Add instructions on how to deploy docs pages to github pages from a new repo in the Scipp organisation

### CODE REVIEW

File: docs/developer/deployment.rst [hunk 1]
```diff
@@ -50,3 +50,35 @@ Mantid Framework Deployment Procedure
...
+
+In this example, we assume that the new repository is named ``ess`` (the full path including organisation is hence ``scipp/ess``).
+
+#. Clone the ``ess`` repository: ``git clone git@github.com:scipp/ess.git``
+
+#. Create a `gh-pages` branch on the repository with ``git checkout -b gh-pages``
+
+#. On Github, go to the repository ``Settings > Pages`` (you will need admin rights to see the ``Settings`` tab). You should see that it is now saying "Your site is published at ``https://scipp.github.io/ess/``". It should also be saying it is using the ``gh-pages`` branch and the ``/root`` folder.
+
+#. On your local machine, generate a ssh key, without an email nor passphrase: ``ssh-keygen -t rsa -b 4096``. When prompted for a file name, use ``azure-ess-key`` (where ``ess`` is the name of the github repository).
```
Comment (OwenArnold) [inline docs/developer/deployment.rst:66]:
```suggestion
#. On your local machine, generate a ssh key, without an email or passphrase: ``ssh-keygen -t rsa -b 4096``. When prompted for a file name, use ``azure-ess-key`` (where ``ess`` is the name of the github repository).
```

File: docs/developer/deployment.rst [hunk 2]
```diff
@@ -50,3 +50,35 @@ Mantid Framework Deployment Procedure
...
+
+#. Go once again to the repository's ``Settings > Deploy keys``. Add a new key, and copy the contents of the ``azure-ess-key.pub`` file. Remember to also give the key ``write`` access to the repository.
```
Comment (OwenArnold) [inline docs/developer/deployment.rst:68]: You previously only said anything about the Settings/Pages. Just strange as you are implying this is the not the first time you've been to the Deploy keys page.

File: docs/developer/deployment.rst [hunk 3]
```diff
@@ -50,3 +50,35 @@ Mantid Framework Deployment Procedure
...
+
+#. Go to the project page on Azure. Go to ``Pipelines > Library > Secure Files``. Upload the ``azure-ess-key`` (without the ``.pub``) file as a secure file.
```
Comment (OwenArnold) [inline docs/developer/deployment.rst:70]: Maybe clarify that this is the private key of the pair (not that you remove the extension from the public one).

File: docs/developer/deployment.rst [hunk 4]
```diff
@@ -50,3 +50,35 @@ Mantid Framework Deployment Procedure
...
+
+#. On your local machine, to back to where you cloned the ``ess`` git repository. Create a new branch starting from the ``main`` branch: ``git checkout main`` then ``git checkout -b deploy_docs``.
+
+#. In that new branch, create a ``.azure-pipelines/documentation_deploy.yml`` script. The easiest way is to copy one from another repository, e.g. `ess-notebooks <https://github.com/scipp/ess-notebooks/blob/master/.azure-pipelines/documentation_deploy.yml>`_.
+
+#. In that ``.azure-pipelines/documentation_deploy.yml`` file, replace the ``sshPublicKey`` with the contents of azure-ess-key.pub. Change the ``sshKeySecureFile`` to ``azure-ess-key``. Change ``git clone git@github.com:scipp/ess-notebooks`` to ``git clone git@github.com:scipp/ess``.
+
+#. Push the changes to github (``git push origin deploy_docs``) and create a pull request. Hopefully, once the pull request is merged into ``main`` the pages will show up on `https://scipp.github.io/ess <https://scipp.github.io/ess>`_.
```
Comment (OwenArnold) [inline docs/developer/deployment.rst:78]: An alternative is to just change the "branches trigger" in the `main.yaml` and at the same time disable publishing of new packages as a temporary change.

File: docs/developer/deployment.rst [hunk 5]
```diff
@@ -50,3 +50,35 @@ Mantid Framework Deployment Procedure
...
+
+#. On your local machine, generate a ssh key, without an email nor passphrase: ``ssh-keygen -t rsa -b 4096``. When prompted for a file name, use ``azure-ess-key`` (where ``ess`` is the name of the github repository).
+
+#. Go once again to the repository's ``Settings > Deploy keys``. Add a new key, and copy the contents of the ``azure-ess-key.pub`` file. Remember to also give the key ``write`` access to the repository.
+
+#. Go to the project page on Azure. Go to ``Pipelines > Library > Secure Files``. Upload the ``azure-ess-key`` (without the ``.pub``) file as a secure file.
+
+#. On your local machine, to back to where you cloned the ``ess`` git repository. Create a new branch starting from the ``main`` branch: ``git checkout main`` then ``git checkout -b deploy_docs``.
+
+#. In that new branch, create a ``.azure-pipelines/documentation_deploy.yml`` script. The easiest way is to copy one from another repository, e.g. `ess-notebooks <https://github.com/scipp/ess-notebooks/blob/master/.azure-pipelines/documentation_deploy.yml>`_.
```
Comment (OwenArnold) [inline docs/developer/deployment.rst:74]: Should we say something about being careful to understand all the steps? Maybe thats obvious and you outline what should need to be changed in the next step anyway.

File: docs/developer/deployment.rst [hunk 6]
```diff
@@ -50,3 +50,35 @@ Mantid Framework Deployment Procedure
   When running ``conda build`` locally, ensure that ``conda-build`` is up to date (``conda update conda-build``). This can be a source of difference between what is observed on the CI (install fresh into clean conda env) and a potentially stale local environment. You should also ensure that the channel order specified is the same as is applied in the CI for your ``conda build`` command. Refer to order applied in ``conda build`` step in pipeline yaml file. Priority decreases from left to right in your command line argument order. You should also ensure that your local `~/.condarc` file does not prioritise any unexpected/conflicting channels and that flag settings such as ``channel_priority: false`` are not utilised. Note that you can set ``--override-channels`` to your ``conda build`` command to prevent local `.condarc` files getting getting in the way.
 
 
+Documentation on Github Pages
+-----------------------------
+
+The procedure below describes the steps to follow in order to publish documentation pages from a new repository in the Scipp organisation to the scipp.github.io website.
```
Comment (OwenArnold) [inline docs/developer/deployment.rst:56]: Could we expand this to explain that we keep all documentation in the `doc` dir under the repository root and use sphinx to transform this rst into useable documentation formats?

**Review by OwenArnold — CHANGES_REQUESTED** [review]
A few things to improve, but generally captures everything I think we need. Thanks!

File: docs/developer/deployment.rst (hunk 5 above)
Comment (nvaytet) [reply docs/developer/deployment.rst:78]: I think it's ok as it is

**Review by OwenArnold — APPROVED** [review]

**Pull request merged**